tools/generator/program.go

```diff
@@ -142,6 +142,7 @@ func main() {
 
 			if packageVersion != "" {
 				args = append(args, fmt.Sprintf("--package-version='%s'", packageVersion))
+				args = append(args, fmt.Sprintf("--user-agent='Azure-SDK-For-Go/%s services'", packageVersion))
```

Comment:
can we pull more info about the platform, e.g. OS, Arch, Device, runtime version? That would give us insights into where our APIs are being called from.

Reply:
We actually already add that information in here :)
[github.com/Azure/go-autorest/autorest/client.go#L43](https://github.com/Azure/go-autorest/blob/67bde902ef44ecb83280eda2dfcc5b85ccc964a6/autorest/client.go#L43)